libs/os/dir.cpp

```diff
@@ -7,34 +7,28 @@
 
 Directory::Directory(const char* name) {
     this->dir = new QDir(name);
+    QStringList entryList = this->dir->entryList(QDir::Dirs | QDir::Files | QDir::NoDotAndDotDot);
+    for(int i = 0; i < entryList.size(); i++) {
+        QString file_path = entryList.at(i);
```

Comment:
This is unnecessary copy. QString is using hashed pool, but it's not free still. Normal practice is taking reference.